### ISSUE TITLE
Client Workflow

### ISSUE THREAD

**Issue by akoebbe** [happybottoms/coverd#86]
The Client entity needs a status workflow. We can use the yohang/Finite library since some of us are familiar with it. But I'm open to other libraries.

Here is a screenshot of the transitions currently being used in Drupal...

<img width="1070" alt="Screen Shot 2020-01-15 at 10 11 34 PM" src="https://user-images.githubusercontent.com/1044465/72492787-23263880-37e4-11ea-8b7b-b612f8040ad8.png">

We can make the permissions portion another ticket if that makes more sense.

**Comment by ericpoe** [comment]
Depends upon #87

**Comment by ericpoe** [comment]
Closed via #104
